### DISCUSSION TITLE
HookWebpackError: Unknown word

### DISCUSSION THREAD

**Discussion by tofuwu-dev** [tailwindlabs/tailwindcss#9390] [Help] [answered]
HI, I get follow error when run `yarn build`.
I use nextjs and vecel to deploy.
![image](https://user-images.githubusercontent.com/98631674/191615694-eaaef2f5-b038-44d4-a858-05418011af91.png)

dependencies
```typescript
...
"next": "12.3.1"
"react": "18.2.0",
"tailwindcss": "^3.1.8"
"sass": "^1.54.9",
...
```

plz help me!

**Comment by wongjn** [comment]
Seems like a plugin somewhere is unhappy with some syntax… Would you be able to share you project configuration files? Perhaps, for PostCSS, Next, any other relevant configuration related to CSS building. The project works fine in dev?

**Reply by tofuwu-dev** [reply]
Thanks for your reply @wongjn 
Yes, app works good on dev mode. 
Which config file do you need exactly?

**Reply by wongjn** [reply]
Would you be able to share you project configuration files? Perhaps, for PostCSS, Next, any other relevant configuration related to CSS building.

**Reply by tofuwu-dev** [reply]
tailwind.config.js
```typescript
/** @type {import('tailwindcss').Config} */

const defaultTheme = require("tailwindcss/defaultTheme");

module.exports = {
  content: ["src/**/*.{js,jsx,ts,tsx}"],
  important: true,
  // prefix: "tw-", // prefix for whether class is tailwind utility class or class for custom style
  theme: {
    extend: {
      colors: {
        bar: "#00D2C8",
      },
      backgroundImage: (theme) => ({
        // primary_gradient: `linear-gradient(to right, ${theme("colors.gradient1")}, ${theme("colors.gradient2")})`,
        card_gradient1: "linear-gradient(180deg, color1 0%, color2 100%)",
        text_gradient2: "linear-gradient(90.12deg, color4 0.05%, color3 70.33%);",
      }),
      boxShadow: {
        button_hover: "0px 7px 10px #123123",
      },
      fontFamily: {
        sans: ["Encode Sans", ...defaultTheme.fontFamily.sans], // default font style
      },
    },
  },
  plugins: [],
};
```

postcss.config.js
```typescript
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```

next.config.js
```typescript
/** @type {import('next').NextConfig} */

const path = require("path");

const nextConfig = {
  experimental: {
    newNextLinkBehavior: true,
  },
  images: {
    formats: ["image/avif", "image/webp"],
  },
  reactStrictMode: true,
  sassOptions: {
    includePaths: [path.join(__dirname, "styles")],
  },
  swcMinify: true,
};

module.exports = nextConfig;
```

**Reply by tofuwu-dev** [reply]
Have you ever had such an issue? Do you have any ideas? Thanks in advance!

**Reply by wongjn** [reply]
No, I haven't had such an issue.

I think it seems the error could be from `css-minimizer` Next.js Webpack plugin – it is mentioned in the error messages, and also I expect it is run only in production build and not in dev which is why your dev build runs fine. All I can suggest at this point is to use a JavaScript debugger and step through the code, perhaps try to take a look at the what precise input is causing the `css-minimizer` to error.

**Comment by tofuwu-dev** [comment, marked as answer]
I was able to figure it out after many tries.
I have read this [issue](https://solveforum.com/forums/threads/solved-why-am-i-getting-this-error-tailwind-next-js-hookwebpackerror-csssyntaxerror.621891/) and googled so many times.
@icyJoseph hinted that need to check tailwind config js file and I could check it out and solved it finally.
I will share it now and hope it would be useful for such an issue.

```typescript
...
 backgroundImage: (theme) => ({
    // primary_gradient: `linear-gradient(to right, ${theme("colors.gradient1")}, ${theme("colors.gradient2")})`,
    card_gradient1: "linear-gradient(180deg, color1 0%, color2 100%)",
    text_gradient2: "linear-gradient(90.12deg, color4 0.05%, color3 70.33%);",
  }),
...
```

`;`, it is issue - text_gradient2: "linear-gradient(90.12deg, color4 0.05%, color3 70.33%)`;`"
Do you see what I was wrong? It was very easy after I found it but I spent plenty of times for it. :)

**Reply by HimayunKhan** [reply]
how you figure where the problem is?? i am not getting where to check the mistake in my project

**Reply by wongjn** [reply]
Consider providing more details on the issue you are experiencing. Consider perhaps creating a new discussion topic, with sufficient debugging details such that we can troubleshoot any unexpected behavior.

**Reply by SupakornNetsuwan** [reply]
I have faced the issue long time ago, so I couldn't even remembered when. Then I am visiting this answer again in 2024 🤣

**Comment by ezzabuzaid** [comment]
In my case I had semicolon in one of tailwind settings

```
extends: {
      boxShadow: {
        main: `rgba(62, 62, 62, 0.04) 0px -2.4px 0px 0px inset;`,
      }
}
```

**Comment by emonno13** [comment]
In my case, when you use "! important" in the tailwind config file, you need to remove the semicolon in the inline tailwind style

<img width="1343" alt="Screenshot 2024-09-25 at 11 06 23" src="https://github.com/user-attachments/assets/fe54a8e9-8448-4752-b883-1b317a26dd03">
<img width="403" alt="Screenshot 2024-09-25 at 11 06 51" src="https://github.com/user-attachments/assets/4a914a40-13aa-4ba3-81f7-0f8b36ee4bac">
